### DISCUSSION TITLE
replace   ntfromstruct()   with  NamedTuple  method

### DISCUSSION THREAD

**Discussion by Lincoln-Hannah** [JeffreySarnoff/NamedTupleTools.jl#43] [Ideas]
```
using WhereTraits

@traits NamedTuple( x ) where isstructtype(typeof(x)) = ntfromstruct( x )
```

The functionality in WhereTraits.jl would allow `ntfromstruct()` to be replaced with a `NamedTuple()`  method.
This reduces the number of interface functions, and seems nice to me.
But I'm not an expert.

**Comment by JeffreySarnoff** [comment]
I am avoiding overloading `NamedTuple` directly.  It is a type exported from Base, so altering its behavior (or adding a constructor) is considered [type piracy](https://discourse.julialang.org/t/clarification-on-type-piracy/5926).   I appreciate your focus on this, thanks.
